### PR TITLE
[SYCL] Ignore vec_type_hint attribute in SYCL 2020

### PR DESCRIPTION
**Pull request by elizabethandrews**
According to the SYCL 2020 spec, [[sycl::vec_type_hint()]] attribute should accept arguments of the type sycl::vec type. The attribute should also be accepted with non conforming lambda syntax.

The current implementation in SYCL corresponds to the openCL version of this argument (with an additional spelling for SYCL), i.e. the attribute accepts extended vector type, floating point types and integral type. An error diagnostic is thrown for sycl:vec type. 

Since the attribute is deprecated and is not handled by any SYCL backend, and will be removed in a future version of the spec, to be minimally conformant with SYCL 2020 spec, this PR just ignores the attribute instead of adding support for sycl::vec type. Support was also added for non conforming lambda syntax

### CODE REVIEW

**Comment by elizabethandrews** [comment]
Just realized I also need to accept this attribute on lambda syntax like other kernel attributes do, to be conformant. I will update PR with a new commit

**Comment by elizabethandrews** [comment]
ping @premanandrao

**Comment by elizabethandrews** [comment]
ping @premanandrao @Fznamznon

File: clang/include/clang/Basic/DiagnosticSemaKinds.td [hunk 1]
```diff
@@ -11979,6 +11979,9 @@ def warn_ivdep_attribute_argument : Warning<
 def warn_attribute_spelling_deprecated : Warning<
   "attribute %0 is deprecated">,
   InGroup<DeprecatedAttributes>;
+def warn_attribute_deprecated_ignored : Warning<
+  "attribute %0 is deprecated; attribute ignored">,
```
Comment (Fznamznon) [inline clang/include/clang/Basic/DiagnosticSemaKinds.td:11983]: We could re-use `warn_attribute_ignored` instead, but that would be two warnings for one attribute, so I don't have a strong opinion here.

File: clang/test/SemaSYCL/vec-type-hint-2.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// RUN: %clang_cc1 -fsycl-is-device -sycl-std=2020 -internal-isystem %S/Inputs -fsyntax-only -verify %s
+
+// Test which verifies [[sycl::vec_type_hint()]] is accepted
+// with non-conforming lambda syntax.
+
+// NOTE: This attribute is not supported in the SYCL backends.
+// To be minimally conformant with SYCL2020, attribute is
+// accepted by the Clang FE with a warning. No additional
+// semanatic handling or IR generation is done for this
```
Comment (Fznamznon) [inline clang/test/SemaSYCL/vec-type-hint-2.cpp:9]:
```suggestion
// semantic handling or IR generation is done for this
```

**Review by Fznamznon — COMMENTED** [review]
A couple of NITs, otherwise LGTM.

File: clang/include/clang/Basic/DiagnosticSemaKinds.td (hunk 1 above)
Comment (elizabethandrews) [reply clang/include/clang/Basic/DiagnosticSemaKinds.td:11983]: The spec authors preferred a single warning. So I think this shoul dbe kept as-is

File: clang/test/SemaSYCL/vec-type-hint-2.cpp (hunk 2 above)
Comment (elizabethandrews) [reply clang/test/SemaSYCL/vec-type-hint-2.cpp]: Thanks for catching this! Fixed.

**Review by Fznamznon — APPROVED** [review]

**Comment by elizabethandrews** [comment]
@intel/llvm-gatekeepers PR is ready for merge

**Pull request merged**